api_calls_string.py

```diff
@@ -0,0 +1,455 @@
+                response = connection.getresponse()
+
+                if response.status != 200:
+                    print("Error: ", response.status, response.reason)
+                    return None
+                
+                data = response.read()
+                
+                json_data = json.loads(data.decode("utf-8"))
+
+               
+                for player in json_data['players']:
+                    writer.writerow([player['full_name'],player['id'], team_id])
+                        
+            # Catching exceptions #
+        except json.JSONDecodeError as e:
+            print(f"A JSONDecodeError occurred: {str(e)}")
+        except http.client.HTTPException as e:
+            print(f"An exception occurred: {str(e)}")
+        except Exception as e:
+            print(f"An exception occurred: {str(e)}")
+        return None
+
+def get_player_stats_from_id(player_id, team_id):
+    try:
+        #Connection & response status
+        comConnect.request("GET", f"/nba/trial/stream/en/statistics/subscribe?api_key={api_key}&players==sd:player:{player_id}")
+
+        response = connection.getresponse()
+        
+        if response.status != 200:
+            print("Error:", response.status, response.reason)
+            return None
+
+        data = response.read()
+        #print(data) #----------------------------------------------------------If the code doesn't work - but the api server isn't idle - could you send the result of this?
+        json_data = json.loads(data.decode("utf-8"))
+
+        # setup of output
+        output = "Name: " + json_data['full_name']
+
+        # gets team name
+        team_name = ""
+        with open(team_id_list, 'r') as file:
+            teams = csv.reader(file)
+            for team in teams:
+                if team_id == team[0]:
+                    team_name = f"{team[2]} {team[1]}"
+
+        output +=" Team: " + team_name
+
+
+        # gets all the relevant stats, and adds it to the output
+        data = json_data['statistics']
+        
+        FGM = data['field_goals_made']
+        FGA = data['field_goals_att']
+        FG = per(FGM,FGA)
+
+        ThreePM = data['three_points_made']
+        ThreePA = data['three_points_att']
+        TP = per(ThreePM,ThreePA)
+
+        FTM = data['free_throws_made']
+        FTA = data['free_throws_att']
+        FT = per(FTM,FTA)
+
+        REB = data['rebounds']
+        AST = data['assists']
+        BLK = data['blocks']
+        STL = data['steals']
+        PTS = data['points']
+
+        output += f" FG: {FG}"
+        output += f" TP: {TP}"
+        output += f" FT: {FT}"
+        output += f" REB: {REB}"
+        output += f" AST: {AST}"
+        output += f" BLK: {BLK}"
+        output += f" STL: {STL}"
+        output += f" PTS: {PTS}"
+
+        return output
+                    
+                    
+                    
+    # Catching exceptions #

```

Comment:
I would add an exception for CSV files on each function that uses csv files to reduce crashing errors.